src/Components/test/E2ETest/Tests/WebAssemblyStringComparisonTest.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    public class WebAssemblyStringComparisonTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        public WebAssemblyStringComparisonTest(
+            BrowserFixture browserFixture,
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+        }
+
+        [Fact]
+        public void InvariantCultureWorksAsExpected()
+        {
+            Navigate(ServerPathBase, noReload: false);
+            Browser.MountTestComponent<StringComparisonComponent>();
+
+            var result = Browser.Exists(By.Id("results"));
+
+            Assert.Equal("Ordinal: False Invariant: True", result.Text);
```

Comment:
I don't fully get how this test is passing *before* we take the Mono update that includes collation data by default. If this somehow works even without the Mono update, does that mean we should be using some other test case that requires the update?

Reply:
My guess is that the collation data was included in the previous Mono update, and the one we were waiting for just changes details about how it works as per all the discussion on the Mono repo. If you have the full story I'd like to know :)

Reply:
We'd picked up some of the Mono changes and I rebased this PR on top of that.